crates/mbe/src/lib.rs

```diff
@@ -268,7 +268,7 @@ impl MacroDef {
 }
 
 impl Rule {
-    fn parse(src: &mut TtIter, expect_arrow: bool) -> Result<Rule, ParseError> {
+    fn parse(src: &mut TtIter, expect_arrow: bool) -> Result<Self, ParseError> {
```

Comment:
The original version looks more readable to me.

Reply:
Oh sorry, I forgot to revert this one